pkg/dns/dns.go

```
@@ -91,7 +64,7 @@ func newHandler(truncate, IPv6 bool, hosts map[string]string, upstreamServers []
 			}
 		}
 	} else {
-		if cc, err = newStaticClientConfig(upstreamServers); err != nil {
+		if cc, err = newStaticClientConfig(opts.UpstreamServers); err != nil {
```

Silent fallback isn't great, but I guess we can leave it along for now (and see what upstream thinks).